File: stacks/gin-app/schemas/order.go
package schemas

import "time"

type OrderItemInput struct {
	ProductID *uint `json:"product_id"`
	Quantity  *int  `json:"quantity"`
}

type OrderInput struct {
	UserID *uint            `json:"user_id"`
	Items  []OrderItemInput `json:"items"`
}

type OrderStatusInput struct {
	Status string `json:"status"`
}

type OrderUserResponse struct {
	ID   uint   `json:"id"`
	Name string `json:"name"`
}

type OrderItemResponse struct {
	ProductID uint    `json:"product_id"`
	Quantity  int     `json:"quantity"`
	UnitPrice float64 `json:"unit_price"`
}

type OrderResponse struct {
	ID        uint                `json:"id"`
	User      OrderUserResponse   `json:"user"`
	Items     []OrderItemResponse `json:"items"`
	Total     float64             `json:"total"`
	Status    string              `json:"status"`
	CreatedAt time.Time           `json:"created_at"`
}

File: stacks/express-app/src/models/order.js
function serializeOrder(order) {
  return {
    id: order.id,
    user: {
      id: order.user.id,
      name: order.user.name
    },
    items: order.items.map((item) => ({
      product_id: item.product_id,
      quantity: item.quantity,
      unit_price: item.unit_price
    })),
    total: order.total,
    status: order.status,
    created_at: order.created_at
  };
}

module.exports = {
  serializeOrder
};
